packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -43,25 +44,45 @@ const String minSdkVersion = '16';
 const String targetSdkVersion = '31';
 const String ndkVersion = '23.1.7779620';
 
-final RegExp _androidPluginRegExp = RegExp(r'com\.android\.tools\.build:gradle:(\d+\.\d+\.\d+)');
+// Update this when new versions of Gradle come out.
+const String _maxKnownAndSupportedGradleVersion = '8.0';
```

Comment:
Should it also be updated for minor versions like 8.0.2? Might be worth including this info in the comment.